src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/taglets/SnippetTaglet.java

```diff
@@ -63,6 +64,38 @@
  */
 public class SnippetTaglet extends BaseTaglet {
 
+    public enum Language {
```

Comment:
The idea of a `Language` object is good, but using an `enum` precludes easy extensibility.

Reply:
We'll figure out the extensibility when the need comes. The language of a snippet defines both the comment marker and the comment type. For now, an alternative to that enum would be a boolean flag, which is less readable.

Reply:
All I'm suggesting is to make `Language` a class, with a public constructor and (for now) some standard static members.  In other words, plan for the possibility of additional languages, by not limiting the code to an enum at this time.

Reply:
Let's figure out the `CoarseParser` and "plain text" issues first and then circle back to this discussion. That said, we need to be mindful of time since we're close to RDP 1.